### PR TITLE
Docs/kotlin coroutines run test

### PR DESCRIPTION
**Pull request by HyungGeun94**
---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://docs.junit.org/snapshot/) and [Release Notes](https://docs.junit.org/snapshot/release-notes.html)

### CODE REVIEW

File: documentation/modules/ROOT/pages/writing-tests/test-classes-and-methods.adoc [hunk 1]
```diff
@@ -56,7 +56,10 @@ following example.
 include::example$java/example/MyFirstJUnitJupiterRecordTests.java[tags=user_guide]
 ----
 
-Test and lifecycle methods may be written in Kotlin and may optionally use the `suspend`
+[[test-classes-and-methods-kotlin-coroutines]]
```
Comment (marcphilipp) [inline documentation/modules/ROOT/pages/writing-tests/test-classes-and-methods.adoc]: There's no need to prefix these IDs anymore now that the documentation is split up into separate pages.

```suggestion
[[kotlin-coroutines]]
```

File: documentation/modules/ROOT/pages/writing-tests/test-classes-and-methods.adoc [hunk 2]
```diff
@@ -71,3 +74,18 @@ https://central.sonatype.com/artifact/org.jetbrains.kotlin/kotlin-reflect[`kotli
 and
 https://central.sonatype.com/artifact/org.jetbrains.kotlinx/kotlinx-coroutines-core[`kotlinx-coroutines-core`]
 to be present on the classpath or module path.
+
+Traditionally, one would use `runBlocking` to bridge the gap between tests and coroutines. However,
```
Comment (marcphilipp) [inline documentation/modules/ROOT/pages/writing-tests/test-classes-and-methods.adoc]: I think we should point out that `runBlocking` is what is done by JUnit behind the scenes for `suspend` lifecycle/test methods. Using `runTest` is an alternative to that.

Comment (HyungGeun94) [reply documentation/modules/ROOT/pages/writing-tests/test-classes-and-methods.adoc]: Thanks for the review! 
I've addressed your feedback. 

I updated the section ID and clarified that JUnit uses runBlocking internally, 
while mentioning runTest as the alternative for skipping delays and controlling virtual time.

File: documentation/src/test/kotlin/example/KotlinCoroutinesDemo.kt [hunk 3]
```diff
@@ -37,5 +38,9 @@ class KotlinCoroutinesDemo {
     @Test
     suspend fun coroutineTest() {
     }
+
+    @Test
+    fun runTest() = runTest {
+    }
```
Comment (marcphilipp) [inline documentation/src/test/kotlin/example/KotlinCoroutinesDemo.kt]:
```suggestion
	// end::user_guide[]
	
	// tag::run-test[]
    @Test
    fun runTest() = runTest {
    }
    // end::run-test[]
    // tag::user_guide[]
```

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
@HyungGeun94 Thank you for your contribution! :+1:

**Comment by HyungGeun94** [comment]
@marcphilipp 

I'm really happy that my first contribution has been merged! 
Thank you for reviewing and guiding me through the process. 
I really appreciate your help. 😊
